docs/reference-guides/best-practices/rancher-server/tips-for-scaling-rancher.md

```diff
@@ -0,0 +1,50 @@
+* Only use custom roles if necessary
+* Use as few rules as possible in custom roles
+* Consider whether adding a role to a user is redundant
+* Consider that using less, but more powerful, clusters may be more efficient
+* Monitoring the rolebindings created from creating new projects for specific purposes versus creating new clusters
+
+### Using Apps V2 Over Apps V1
+There are two App kubernetes resources that Rancher uses: apps.projects.cattle.io and correct this apps.cattle.cattle.io. V1 here refers to apps.projects.cattle.io since they were introduced first and are legacy, while V2 refers to apps.catalog.cattle.io. Apps V2 are preferrable because they live in the downstream cluster while Apps V1 lives in the local cluster. The dashboard uses Apps V2 while the old `/g/clusters` UI uses Apps V1.
+
+### Using the Authorized Cluster Endpoint (ACE)
+There is an option for RKE cluster called Authorized Cluster Endpoint. When enabled this adds a context to kubeconfigs generated for the cluster that uses a direct endpoint to the cluster. It is not enough to enable this option. The user of the config needs to use `kubectl use-context <ace context name>` in order to start using it.
+
+Without using ACE, all kubeconfig requests first route through Rancher. This contributes to load on the Rancher setup, while using ACE completely bypasses the Rancher setup.
+
+### Option to Reduce Event Handler Executions
+The bulk of Rancher's logic are ran by event handlers. These event handlers run on an object whenever the object is updated. They also run when Rancher is started. By default they run every 15 hours when caches are synced but this can be changed using the CATTLE_SYNC_ONLY_CHANGED_OBJECTS environment variable. If resource allocation spikes are seen on an interval of about 15 hours it is possible this setting can help. The value for the environment variable can be a comma separated list of the following options: mgmt, user, and scaled. The values refer to types of controllers (the structures that contain and run handlers) and their handlers. Mgmt refers to management which only runs on one Rancher node. User will run for every cluster. Some user handlers run on a Rancher node while others run downstream in their respective cluster. Scaled runs on every Rancher node. The two options that cover most controllers' handlers and are supported are mgmt and user. There are very few "scaled" controllers and their handlers are often critical. It is not recommended that the scaled value is used.
+
+## Optimizations Outside of Rancher
+A large component of performance is the local cluster and how it was configured. Since Rancher is "Bring your own cluster", this impactful component introduces a potential bottleneck before Rancher software ever runs. When Rancher nodes experience high resource usage, you can use the command "top" to identify whether it is Rancher or a kubernetes component that is consuming the resource in excess.
```

Comment:
```suggestion
A large component of performance is the local cluster and how it was configured. This cluster can introduce a bottleneck before Rancher software ever runs. When Rancher nodes experience high resource usage, you can use the command "top" to identify whether it is Rancher or a Kubernetes component that is consuming the resource in excess.
```

Reply:
I suppose this is fine for now but in the future I would like to dive more into how the user brings the cluster and what that entails if they want a performant system.